rust/agama-server/src/storage/web.rs

```diff
@@ -150,24 +207,79 @@ async fn product_params(
     Ok(Json(params))
 }
 
-async fn usable_devices(State(state): State<StorageState<'_>>) -> Result<Json<Vec<String>>, Error> {
-    let devices = state.client.available_devices().await?;
-    let devices_names = devices.into_iter().map(|d| d.name).collect();
+#[derive(Debug, Clone, Serialize, utoipa::ToSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct ProductParams {
+    /// Mount points defined by the product.
+    mount_points: Vec<String>,
+    /// Encryption methods allowed by the product.
+    encryption_methods: Vec<String>,
+}
 
-    Ok(Json(devices_names))
+/// Gets the actions to perform in the storage devices.
+#[utoipa::path(
+    get,
+    path = "/proposal/actions",
+    context_path = "/api/storage",
+    responses(
+        (status = 200, description = "List of actions", body = Vec<Action>),
+        (status = 400, description = "The D-Bus service could not perform the action")
+    )
+)]
+async fn actions(State(state): State<StorageState<'_>>) -> Result<Json<Vec<Action>>, Error> {
+    Ok(Json(state.client.actions().await?))
+}
+
+/// Gets the SID (Storage ID) of the devices usable for the installation.
```

Comment:
np: it looks weird having two "names" for the same thing: "Storage ID" vs "Device ID". But we can keep it as it is.

Reply:
Maybe we could rename `DeviceSid` as `StorageId`. Anyway, I would even prefer to simply have `id` property in the devices, hidding the `sid` concept (even in the D-Bus layer).

Reply:
Sure, no problem, just a comment.